src/components/AddItemModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import DefaultModal, { Overlay, ButtonField } from './DefaultModal';
```

Comment:
As mentioned in another comment, I think we should make Overlay the top level component of the default modal. As for the button field, I think if it is generic enough to use over again, we should consider just passing in the two buttons, the submit button and the cancel button, otherwise it can be put in the AddItemModal component.

Reply:
Sorry; do you mean `DefaultModal` should take two buttons as props and then render them in the footer automatically?